svm-transaction/src/nonce_extraction.rs

```diff
@@ -2,16 +2,12 @@
 //!
 
 use {
-    solana_program::program_utils::limited_deserialize,
-    solana_sdk::{
-        nonce::NONCED_TX_MARKER_IX_INDEX, pubkey::Pubkey, system_instruction::SystemInstruction,
-        system_program,
-    },
-    solana_svm_transaction::svm_message::SVMMessage,
+    crate::svm_message::SVMMessage,
+    solana_sdk::{nonce::NONCED_TX_MARKER_IX_INDEX, pubkey::Pubkey, system_program},
 };
 
 /// If the message uses a durable nonce, return the pubkey of the nonce account
-pub fn get_durable_nonce(message: &impl SVMMessage) -> Option<&Pubkey> {
+pub fn get_durable_nonce(message: &(impl SVMMessage + ?Sized)) -> Option<&Pubkey> {
```

Comment:
notation here is a bit odd; because we essentially have the trait calling into itself for the default implementation.